index.bs

```diff
@@ -59,6 +60,8 @@ different storage mechanism with a different API for such files. The entry point
 
 A <dfn export id="entry">file system entry</dfn> is either a [=file entry=] or a [=directory entry=].
 
+<!-- TODO: Consider tying access checks to a path, to support handles which don't map to an entry -->
```

Comment:
fwiw, at least originally we intentionally did not tie access checks to paths, to provide more flexibility for implementations (and I don't think even today our implementation actually ties access checks to path either?). One case where this makes a difference is:
- get a read-only directory handle via a directory picker
- get a writable file handle, that happens to be in the above directory
- now get a file handle to the same file by calling getFileHandle on the directory handle

I'm pretty sure that in at least our implementation the latter file handle will still be read-only. (i.e. while we sort of tie access checks to paths, they're not tied to the path of the entry a handle refers to, rather they are tied to the path of the file or directory that a user at some point picked).

Reply:
You're correct that, at least in Chrome's current implementation, the file handle you get from calling `getFileHandle()` on the directory handle is read-only, though it's debatable whether that's a feature or a bug. I can see this leading to confusion on the part of the website author and user. Regardless, that's a Chrome implementation detail...

The reason I mention this is because if we assert that an entry must exist on disk, then we can't use it to lock/check access to entries that don't exist, such as when locking the target file when renaming a file.

... thinking about this a bit more, perhaps the `move()` case isn't actually an issue if we specify that all file system operations to the same queue (#95), since that would guarantee (at least within the OPFS) that the target file isn't created and locked while the `move()` operation is in progress

Not sure what to do about access checks, though, if we assert that an "entry" exists on disk